bindings/src/main/java/io/pravega/storage/filesystem/FileSystemStorage.java

```diff
@@ -187,6 +189,34 @@ public boolean supportsTruncation() {
         return false;
     }
 
+    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings(value = "OS_OPEN_STREAM", justification = "Rare operation. The leaked object is collected by GC. In case of a iteraror in a for loop this would be fast.")
+    @Override
+    public Iterator<SegmentProperties> listSegments() {
+        try {
+            return Files.find(Paths.get(config.getRoot()),
+                    Integer.MAX_VALUE,
+                    (filePath, fileAttr) -> fileAttr.isRegularFile())
+                    .map(path -> (SegmentProperties) getStreamSegmentInformation(config.getRoot(), path))
+                    .iterator();
+        } catch (IOException e) {
+            log.error("Exception occurred while listing the segments.", e);
```

Comment:
You should throw this exception. If you just return an empty iterator then the caller will never know something went amiss.